examples/stable/test/property/ic_api/crypto_get_random_values/test/tests.ts

```diff
@@ -0,0 +1,167 @@
+
+                                    await validateCryptoGetRandomValues(
+                                        actor,
+                                        name,
+                                        bytesPerElement,
+                                        length,
+                                        numCalls
+                                    );
+                                }
+                            ),
+                            defaultPropTestParams()
+                        );
+                    });
+
+                    it(`should throw quota exceeded error for ${name} with byte length > 65_536`, async () => {
+                        const actor = await getCanisterActor<Actor>('canister');
+                        const minElementsToExceed =
+                            Math.floor(65_536 / bytesPerElement) + 1;
+
+                        await fc.assert(
+                            fc.asyncProperty(
+                                fc.integer({
+                                    min: minElementsToExceed,
+                                    max: 1_000_000
+                                }),
+                                async (length) => {
+                                    await expect(
+                                        actor.cryptoGetRandomValues(
+                                            name,
+                                            length
+                                        )
+                                    ).rejects.toThrow(
+                                        'QuotaExceeded: array cannot be larger than 65_536 bytes'
+                                    );
+                                }
+                            ),
+                            defaultPropTestParams()
+                        );
+                    });
+                }
+            );
+        });
+    };
+}
+
+async function validateCryptoGetRandomValues(
+    actor: ActorSubclass<Actor>,
+    name: string,
+    bytesPerElement: number,
+    length: number,
+    numCalls: number
+): Promise<void> {
+    const randomResults = await Promise.all(
+        Array(numCalls)
+            .fill(0)
+            .map(() => actor.cryptoGetRandomValues(name, length))
+    );
+
+    randomResults.forEach((randomResult) => {
+        expect(randomResultIsUnique(randomResult)).toBe(true);
+
+        const expectedByteLength = length * bytesPerElement;
+
+        expect(randomResult.length).toBe(expectedByteLength);
+
+        for (const byte of randomResult) {
+            expect(byte).toBeGreaterThanOrEqual(0);
+            expect(byte).toBeLessThanOrEqual(255);
+        }
+    });
+}
+
+let pastRandomResults: Set<string> = new Set();
+
+function randomResultIsUnique(randomResult: Uint8Array | number[]): boolean {
+    if (randomResult.length === 0) {
+        return true;
+    }
+
+    const sizeBefore = pastRandomResults.size;
+    pastRandomResults.add(Buffer.from(randomResult).toString('hex'));
```

Comment:
Maybe this is impossible, I'm just trying to think of any reason that this could be wrong, but say that our implementation was only slightly random and after a say 5 bytes it started repeating itself. and our sample size was 4, we would get [1, 2, 3, 4] then [5, 1, 2, 3] then [4, 5, 1, 2, 3] and there would be repeating and it's not technically unique but because it's only looking to make sure the whole array is unique we don't catch it? I don't know, this seems pretty outlandish to me that this would be a problem.